apps/nestjs-api/src/reminder-emails/reminder-emails.controller.ts

```diff
@@ -175,23 +175,46 @@ export class ReminderEmailsController {
     }
   }
 
-  // @Get('/s3cr3t-3ndp01nt-t0-tr1gg3r-r3m1nd3r5/mentoring-sessions-logging')
-  // async sendMentoringSessionsLoggingReminder() {
-  //   const conProfilesWithMentoringSessionsToLog =
-  //     await this.reminderEmailsService.getConProfilesWithMentorshipMatchesWithoutMentoringSessions(
-  //       { mentorshipMatchAgeInDays: 402 }
-  //     )
-
-  //   if (conProfilesWithMentoringSessionsToLog.length > 0) {
-  //     // send reminder emails
-  //     conProfilesWithMentoringSessionsToLog.forEach(async (profile) => {
-  //       await this.reminderEmailsService.sendLogMentoringSessionsReminder({
-  //         email: profile.props.email,
-  //         firstName: profile.props.firstName,
-  //         userType: profile.props.userType,
-  //         mentorshipMatchAgeInDays: 14,
-  //       })
-  //     })
-  //   }
-  // }
+  @Get('/s3cr3t-3ndp01nt-t0-tr1gg3r-r3m1nd3r5/mentoring-sessions-logging')
+  async sendMentoringSessionsLoggingReminder() {
+    const mentorshipMatchesWithMentorAndMenteeDetails_2weeks =
+      await this.reminderEmailsService.getMentorsAndMenteesWithoutMentoringSessionsLogged(
+        { mentorshipMatchAgeInDays: 14 }
+      )
+
+    // TODO: Uncomment and merge this two weeks after the first reminder is sent
+    // const mentorshipMatchesWithMentorAndMenteeDetails_4weeks =
+    //   await this.reminderEmailsService.getMentorsAndMenteesWithoutMentoringSessionsLogged(
+    //     { mentorshipMatchAgeInDays: 28 }
+    //   )
+
+    if (
+      Object.keys(mentorshipMatchesWithMentorAndMenteeDetails_2weeks).length > 0
+    ) {
+      // send reminder emails
+      Object.keys(mentorshipMatchesWithMentorAndMenteeDetails_2weeks).forEach(
```

Comment:
Doesn't `Object.keys(mentorshipMatchesWithMentorAndMenteeDetails_2weeks)` return an array of strings?

Reply:
Yes, then if you check two lines later, I do something like this:

`mentorshipMatchesWithMentorAndMenteeDetails_2weeks[key]`

This is because for some reason, the lodash function I use to get `mentorshipMatchesWithMentorAndMenteeDetails` (transform) doesn't return an array but kind of an object. I didn't have time to dig deeper in that, but I noticed I cannot just do `mentorshipMatchesWithMentorAndMenteeDetails_2weeks.forEach` so I have to first iterate the keys, then use the key to obtain the value.